lms/djangoapps/discussion/rest_api/serializers.py

```diff
@@ -425,6 +426,24 @@ def update(self, instance, validated_data):
             requesting_user_id = self.context["cc_requester"]["id"]
             if key == "closed" and val:
                 instance["closing_user_id"] = requesting_user_id
+                event_data = {
+                    'request': self.context['request'],
+                    'course': self.context['course'],
+                    'thread': instance,
+                    'close_reason_code': validated_data.get('close_reason_code')
+                }
+                track_thread_lock_unlock_event(**event_data)
+
+            if key == "closed" and not val:
+                instance["closing_user_id"] = requesting_user_id
+                event_data = {
+                    'request': self.context['request'],
+                    'course': self.context['course'],
+                    'thread': instance,
+                    'close_reason_code': validated_data.get('close_reason_code')
+                }
+                track_thread_lock_unlock_event(**event_data, locked=False)
```

Comment:
Can we move all this logic inside the event method as well, and keep the serializer clean?
We should be only calling the thread method here, and compiling the event data inside the event method

Reply:
Here `event_data` is basically a param for that method so it is already in `track_thread_lock_unlock_event` other than ` if key == "closed" and not val:` cannot be moved inside the method because these values are in loop and are conditional .

Reply:
We can call the tracking method 2 times.
We can move the `event_data` dict to the event method. We would have something like this.

ON LINE#429
```python
track_thread_lock_unlock_event(self.context['request'], self.context['course'], instance, validated_data.get('close_reason_code'))
```

ON LINE#439
```python
track_thread_lock_unlock_event(self.context['request'], self.context['course'], instance, validated_data.get('close_reason_code'), False)
```

We can add the `event_data` dict within the event function to clean the serializer, instead of creating the context here and passing it in as `**event_data`.